core/spec.md

```diff
@@ -2902,21 +2913,69 @@ The following describes the attributes of Registry model:
+    Version attributes).
+  - These attributes are typically reserved for system-managed attributes
+    such as `metaurl`. User-defined attributes would normally appear on the
+    Versions, and as such will appear in the model under the
+    `groups.resources.attributes` list.
+  - Extension attribute names at this level MUST NOT overlap with extension
+    attributes defined at the `groups.resources.attributes` level.
+    The only duplicate names allowed are specification defined attributes
+    such as `self` and `xid`, and the Version-specific values MUST be
+    overridden by the Resource-level values when serialized.
 
 - `groups."STRING".resources."STRING".metaattributes`
   <span id="model.metaattributes"></span>
   - See [`attributes`](#model.attributes) above.
   - OPTIONAL.
   - The list of attributes associated with the Resource, not its Versions,
-    and will appear in the `meta` sub-object of the Resource.
-  - The list of `groups.resources.attributes` names MUST NOT overlap with the
-    list of `groups.resource.metaattributes` names.
+    that will appear in the `meta` sub-object of the Resource.
+
+Clarifying the  usage of the `attributes`, `resourceattributes` and
+`metaattributes`:
+- The serialization of the Resource is meant to be (almost) interchangeable
+  with the serialization of a Version of that Resource. This will allow users
+  to process either entity without needing to know if they were provided the
+  Resource itself or a specific Version in most cases.
+- To enable this, most of the Resource-specific data (e.g. its
+  `defaultversionid`), is serialized under the `meta` sub-object. This avoids
+  potential name conflicts between Version and Resource-level attributes, as
+  well a avoiding making the serialization of the Resource too verbose/noisy.
+- However, there are some Resource-level attributes, that if placed in the
+  `meta` sub-object, would appear to be misplaced. For example, the `versions`
+  collection attributes could be confusing to users since `meta` is not
+  the direct parent/owner of the "versions" collection, the Resource is.
+  Especially when considering the URL path to access it.
+- Additionally, some common attributes (e.g. `self`) need to appear on both
+  Resources as well a Versions but the values need to be different in each
+  case. This is why the same attribute names can appear both the
+  `resourceattributes` and `attributes` lists, but only specification defined
+  attributes are allow to have this naming conflict. Extensions are not, as
+  that could lead to confusion for users.
+- Finally, in the vast majority of cases it is expected that models will only
+  need to define Version-level attributes, leaving the more advanced uses of
+  Resource and Meta-level attributes to default to the specification defined
+  sets. For this reason, the Version-level attributes use a list called
+  `attributes` as that would make user creation of the model easier, leaving
```

Comment:
```suggestion
  `attributes` in order to make user creation of the model easier, leaving
```

(Because "as that would" sounds like it's something that isn't actually the case. I may have misunderstood though.

Reply:
done- I like it